source/server/http/admin.cc

```diff
@@ -539,9 +539,10 @@ Http::Code AdminImpl::handlerConfigDump(absl::string_view, Http::HeaderMap& resp
     auto& any_message = *(dump.add_configs());
     any_message.PackFrom(*message);
   }
+  const auto redacted = MessageUtil::redact(dump);
```

Comment:
Thanks. Given the limitations that are now spelled out around `Struct`, how comfortable are we using this for config dump in admin endpoint when we have the `TlsCertificate` inside an opaque `TransportSocket` (which is now the recommended way to do it, explicit TLS contexts are deprecated)? Do we need some comments here or user facing documentation on the limits of redaction.

Ultimately I'm wondering how we're going to document the effective result of this process in version history in a way that end users can grok and get predictable behavior from.

Reply:
Yeah, this is tough. On the one hand, usage of `Struct` is deprecated in v2 (including `TransportSocket.config`), and we should be safer in v3. On the other hand, even in v3 we're still not totally safe with this approach: in the case you described at some point earlier, where some extension is dynamically loaded, it's possible that the extension's type urls won't be registered in the descriptor pool.

I think I'd like to move forward with this anyways, just so we can have something in place, but I definitely agree it's dissatisfying having this limitation.